pixl_core/src/core/upload.py

```diff
@@ -11,179 +11,149 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-"""Functionality to upload files to an endpoint."""
+"""Functionality to upload files to a remote server."""
 
 from __future__ import annotations
 
 import ftplib
 import logging
-import ssl
+from abc import ABC, abstractmethod
 from datetime import datetime, timezone
-from ftplib import FTP_TLS
 from pathlib import Path
-from typing import TYPE_CHECKING, Any, BinaryIO
+from typing import TYPE_CHECKING, BinaryIO
 
 from decouple import config
 
 if TYPE_CHECKING:
-    from socket import socket
-
     from core.exports import ParquetExport
+    from core.project_config import PixlConfig
 
 
-from core.db.queries import get_project_slug_from_db, update_exported_at
+from core._secrets import AzureKeyVault
+from core._upload_ftps import (
+    _connect_to_ftp,
+    _create_and_set_as_cwd,
+    _create_and_set_as_cwd_multi_path,
```

Comment:
really odd to be importing private functions, maybe we should rethink this a bit?

Reply:
I'd argue for 
- moving the FTPSUploader child class into the _upload_ftps module 
- making the _upload_ftps module public
- making ImplicitFtpTls a private class

Reply:
Agreed!